protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMessageRangeProcessor.java

```diff
@@ -113,7 +113,9 @@ protected void doProcess(M request, ImapSession session, String tag, ImapCommand
             }
             taggedBad(command, tag, responder, HumanReadableText.INVALID_MESSAGESET);
         } catch (MailboxException e) {
-            LOGGER.error(getOperationName() + " failed from mailbox " + currentMailbox.getPath() + " to " + targetMailbox + " for sequence-set " + ImmutableList.copyOf(idSet), e);
+            if (LOGGER.isErrorEnabled()) {
```

Comment:
What is the point of logger.is\*Enabled() everywhere in this PR?

I know it is to avoid string formatting, but I feel like it might be sometime premature optimization... I don't think this log operation is expensive, and I would value not making the code look more complex... 

Also what forbids us to use "placeholders"?

Reply:
:+1:

Reply:
In addition to string concatenations, the is*Enabled() checks help prevent unnecessary work from being performed if, ultimately, the work will be wasted. Here, for example, there is a call to `ImmutableList.copyOf(idSet)`.

If I wasn't sure if the amount of work could somehow be controlled by an external actor, possibly leading to a DoS vulnerability, I added the check for is*Enabled(). If this concern is not applicable, then parameterized logging is probably the way to go.

The Guava implementation of ImmutableList.copyOf(E[]) generally will create a copy of the array:

https://github.com/google/guava/blob/e3b8c0b6f07c1017e80d86758191d6898517a546/guava/src/com/google/common/collect/ImmutableList.java#L285..L294

Now that I look at the code again, I think we can use Arrays.asList() here because a copy doesn't actually need to be made. The reason is to convert the array to a `List`.

Reply:
I just pushed a commit making this change.

Reply:
I was mistaken in thinking that SLF4J supported logging Collections directly. Actually, it only appeared to work because most implementations of Collection.toString() come from `java.util.AbstractCollection`.

SLF4J "natively" supports printing arrays (see https://github.com/qos-ch/slf4j/blob/a81440ea36676f3b090f5a4ccba70d17c7b80f6e/slf4j-api/src/main/java/org/slf4j/helpers/MessageFormatter.java#L265 ), so I think that it is better to pass `idSet ` directly to the logger call as a formatting arg. I have just pushed a commit making this change.